lib/AST/RequirementMachine/Rule.cpp

```diff
@@ -146,6 +147,14 @@ bool Rule::isCircularConformanceRule() const {
   return true;
 }
 
+/// Returns \c true if this rule is prefixed with the \c [element] symbol.
+bool Rule::isSameElementRule() const {
+  if (LHS.size() == 0)
```

Comment:
LHS cannot be empty, because `LHS > RHS`, and nothing strictly smaller than an empty term. It would also cause an infinite reduction because `(e => RHS)` means you can insert RHS anywhere in the middle of a term, and nothing is reduced :)

Reply:
Fixed!